src/operator/convolution.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+import edgeHandlingMethod from './edgeHandlingMethod'
+
+//convolution using a square kernel
+export default function convolution(newImage, kernel, edgeHandling){
+    let kernelWidth = Math.sqrt(kernel.length);
+    let k = Math.floor(kernelWidth/2);
+    let div = 0;
+    edgeHandling = edgeHandling || 'copy';
+
+    for(let i = 0; i < kernel.length; i++){div += kernel[i];}
+
+    for(let x = 0; x < this.width ; x++){
+        for(let y = 0; y < this.height ; y++){
+            let sum = 0;
+            for(let i = -k; i <= k; i++){
+                for(let j = -k; j <= k; j++){
+                    let val = edgeHandling.toLowerCase() == 'mirror'
```

Comment:
this part is the limiting step in the code, you must avoid as much as possible to use conditions and especially functions like toLowerCase.
The edgeHandling option does not change during the algorithm so it can be checked before starting the loop.
You can do something like that before the first loop:
```js
let edgeHandler;
switch(edgeHandling.toLowerCase()){
  case 'mirror':
    edgeHandler = myMirrorHandler;
    break;
  case 'copy':
      edgeHandler = myCopyHandler;
    break;
   ...
   default:
     throw new Error('unknown edge handler: ' + edgeHandling);
}
```